xdsl/transforms/scf_parallel_loop_tiling.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import dataclass
+
+from xdsl.dialects import affine, arith
+from xdsl.dialects.builtin import IndexType, IntegerAttr, ModuleOp
+from xdsl.dialects.scf import ParallelOp, Yield
+from xdsl.ir import Block, MLContext, Operation, Region, SSAValue
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.hints import isa
+
+
+@dataclass
+class ScfParallelLoopTilingPattern(RewritePattern):
+    tile_sizes: list[int]
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ParallelOp, rewriter: PatternRewriter, /):
+        # Only tile the innermost parallel loop
+        if any(isinstance(o, ParallelOp) for o in op.body.walk()):
+            return
+        lower = op.lowerBound
+        upper = op.upperBound
+        step = op.step
+
+        # fill the tile sizes with ones
+        tile_sizes_v = self.tile_sizes
+        for _ in range(len(tile_sizes_v), len(lower)):
+            tile_sizes_v.append(1)
+
+        # Only handle constant loop bounds for now
+        # if not (
+        #     isa(lower, tuple[arith.Constant, ...])
+        #     and isa(upper, tuple[arith.Constant, ...])
+        #     and isa(step, tuple[arith.Constant, ...])
+        # ):
+        #     return
```

Comment:
Am I right in understanding that the comment is about the commented code? It feels ambiguous, hard to tell whether the comment is related to the commented code or just there... It might be worth filing an issue to resolve this, or add a comment with something like "code commented because ..."

Reply:
Just forgotten commented code, will clean!